src/vm/class.cpp

```diff
@@ -291,11 +291,6 @@ VOID EEClass::FixupFieldDescForEnC(MethodTable * pMT, EnCFieldDesc *pFD, mdField
     }
 #endif // _DEBUG
 
-    // MethodTableBuilder uses the stacking allocator for most of it's
-    // working memory requirements, so this makes sure to free the memory
-    // once this function is out of scope.
-    ACQUIRE_STACKING_ALLOCATOR(pStackingAllocator);
```

Comment:
Do you have an example stacktrace that failed here? The issue we discussed the other day doesn't make much sense in this method because the contract above is MODE_COOPERATIVE which means GetThread() should be a non-NULL value. Code review also indicates that MethodTableBuilder::InitializeFIeldDescs called below will directly use the stacking allocator if a ValueType field is present in the type definition (https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L4122)

Reply:
> the contract above is MODE_COOPERATIVE which means GetThread() should be a non-NULL value

The contracts in EnC code cannot be trusted. EnC has several broad `CONTRACT_VIOLATION` suppressions.

Reply:
Indeed, this resulted in an AV in fixups. Reverted this part of the change.

Reply:
Thanks